checkov/sast/runner.py

```diff
@@ -0,0 +1,168 @@
+        semgrep_output = SemgrepOutput(filtered_matches_by_rule, semgrep_errors, renamed_targets,
+                                       target_manager_ignore_log, filtered_rules, profiler,
+                                       output_extra, shown_severities, target_manager_lockfile_scan_info)
+        return semgrep_output
+
+    def _create_report(self, filtered_matches_by_rule: Dict[Rule, List[RuleMatch]]) -> Report:
+        report = Report(self.check_type)
+        for rule, matches in filtered_matches_by_rule.items():
+            for match in matches:
+                check_id = rule.id.split('.')[-1]
+                check_name = rule.metadata.get('name', '')
+                code_block = Runner._get_code_block(match.lines, match.start.line)
+                file_abs_path = match.match.location.path
+                file_path = file_abs_path.split('/')[-1]
+                severity = get_severity(SEMGREP_SEVERITY_TO_CHECKOV_SEVERITY.get(rule.severity))
+                file_line_range = [match.start.line, match.end.line]
+                check_result = _CheckResult(result=CheckResult.FAILED)
+
+                record = Record(check_id=check_id, check_name=check_name, resource="", evaluations={},
+                                check_class="", check_result=check_result, code_block=code_block,
+                                file_path=file_path, file_line_range=file_line_range,
+                                file_abs_path=file_abs_path, severity=severity)
+                report.add_record(record)
+        return report
+
+    @staticmethod
+    def _get_code_block(lines: List[str], start: int) -> List[Tuple[int, str]]:
+        code_block = []
+        index = start
+        for line in lines:
+            code_block.append((index, line))
+            index += 1
+        return Runner._cut_code_block_ident(code_block)
+
+    @staticmethod
+    def _cut_code_block_ident(code_block: List[Tuple[int, str]]) -> List[Tuple[int, str]]:
+        min_ident = len(code_block[0][1]) - len(code_block[0][1].lstrip())
+        for item in code_block[1:]:
+            current_min_ident = len(item[1]) - len(item[1].lstrip())
+            if current_min_ident < min_ident:
+                min_ident = current_min_ident
+
+        if min_ident == 0:
+            return code_block
+
+        code_block_cut_ident = []
+        for item in code_block:
+            code_block_cut_ident.append((item[0], item[1][min_ident:]))
+        return code_block_cut_ident
+
+    @staticmethod
+    def _get_generic_ast(language: SastLanguages, target: str) -> Dict[str, Any]:
+        try:
+            core_runner = CoreRunner(jobs=None, engine=EngineType.OSS, timeout=DEFAULT_TIMEOUT, max_memory=0,
+                                     interfile_timeout=0, timeout_threshold=0, optimizations="none", core_opts_str=None)
+            cmd = [SemgrepCore.path()] + ['-json', '-full_token_info', '-dump_ast', target, '-lang', language.value]
```

Comment:
```suggestion
            cmd = [SemgrepCore.path(), '-json', '-full_token_info', '-dump_ast', target, '-lang', language.value]
```
why not fusing both together directly?